runblueprint/plans/test/test_planner.py

```diff
@@ -61,3 +67,95 @@ def test_determine_recovery_start_race_week(self):
         start_date = datetime.date(2016, 11, 7)  # Mon
         result = determine_recovery_start(race_date, start_date)
         self.assertEqual(result, datetime.date(2016, 11, 13))  # Should start on Monday after the race
+
+    def test_get_recovery_percent_4(self):
+        total_weeks = 4
+
+        recovery_percents = get_weekly_recovery_percents(total_weeks)
+
+        self.assert_listAlmostEqual([0.2, 0.4, 0.6, 0.8], recovery_percents)
+
+    def test_get_recovery_percent_4(self):
+        total_weeks = 3
+
+        recovery_percents = get_weekly_recovery_percents(total_weeks)
+
+        self.assert_listAlmostEqual([0.25, 0.5, 0.75], recovery_percents)
+
+    def test_get_recovery_percent_1(self):
+        total_weeks = 1
+
+        recovery_percents = get_weekly_recovery_percents(total_weeks)
+
+        self.assert_listAlmostEqual([0.5], recovery_percents)
+
+    def test_get_recovery_percent_0(self):
+        total_weeks = 0
+
+        recovery_percents = get_weekly_recovery_percents(total_weeks)
+
+        self.assert_listAlmostEqual([], recovery_percents)
+
+    def test_get_recovery_percent_negative(self):
+        total_weeks = -1
+
+        recovery_percents = get_weekly_recovery_percents(total_weeks)
+
+        self.assert_listAlmostEqual([], recovery_percents)
+
+    def test_get_recovery_dists_1(self):
+        total_weeks = 1
+        post_recovery_dist = 20.0
+        recovery_dists = get_weekly_recovery_dists(total_weeks, post_recovery_dist)
+
+        self.assert_listAlmostEqual([post_recovery_dist/2.0], recovery_dists)
+
+    def test_get_recovery_dists_2(self):
+        total_weeks = 2
+        post_recovery_dist = 20.0
+        recovery_dists = get_weekly_recovery_dists(total_weeks, post_recovery_dist)
+
+        self.assert_listAlmostEqual([post_recovery_dist/3.0, post_recovery_dist * (2.0/3.0)], recovery_dists)
+
+    def test_get_recovery_dists_negative_weeks(self):
+        total_weeks = -1
+        post_recovery_dist = 20.0
+
```

Comment:
Extra whitespace is a bit hard to read, easier to just keep it as a single paragraph I think.